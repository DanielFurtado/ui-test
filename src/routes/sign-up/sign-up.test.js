import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

import SignUp from './sign-up.component';

describe(('Sign up component'), () => {
  it('renders', () => {
    const component = renderer.create(
      <Router>
        <SignUp />
      </Router>
      );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

