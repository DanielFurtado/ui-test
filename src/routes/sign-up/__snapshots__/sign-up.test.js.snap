// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up component renders 1`] = `
<div
  className="sign-up-container"
>
  <div
    className="sign-up-wrapper"
  >
    <div
      className="sign-up-heading"
    >
      <h1>
        Sign up to our newsletter
      </h1>
      <p>
        Join our mailing list and you'll be the first to hear about our latest updates, news, and special offers.
      </p>
    </div>
    <div
      className="sign-up-form-wrapper"
    >
      <form
        arial-label="Sign up form"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="name"
          >
            Name *
          </label>
          <input
            aria-invalid="false"
            aria-label="name"
            id="name"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Name"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="email"
          >
            Email *
          </label>
          <input
            aria-invalid="false"
            aria-label="email"
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email"
            type="email"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="dob"
          >
            Date of Birth *
          </label>
          <input
            aria-invalid="false"
            aria-label="date of birth"
            id="dob"
            name="dob"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Date of Birth"
            type="date"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="email"
          >
            Select your favourite color:
          </label>
          <input
            aria-label="favorite colour"
            className="color-picker"
            id="favoriteColour"
            name="favoriteColour"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Favorite Colour"
            type="color"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            data-testid="salary-value-display"
            htmlFor="email"
          >
            Salary: Â£ 
            0
          </label>
          <input
            aria-label="salary"
            data-testid="salary-range-input"
            id="salary"
            max="250000"
            min="0"
            name="salary"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Salary"
            step="1000"
            type="range"
            value={0}
          />
        </div>
        <button
          className="button-primary"
          data-testid="signup-submit-button"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
